tools/Azure.Mcp.Tools.AzureManagedLustre/src/Commands/BaseAzureManagedLustreCommand.cs

```diff
@@ -15,4 +17,111 @@ public abstract class BaseAzureManagedLustreCommand<
 {
     // Currently no additional options beyond subscription + resource group
     protected readonly ILogger<BaseAzureManagedLustreCommand<TOptions>> _logger = logger;
+
+    public ValidationResult ValidateRootSquashOptions(CommandResult commandResult, CommandResponse? commandResponse = null)
+    {
+        var rootSquashMode = commandResult.GetValueOrDefault(AzureManagedLustreOptionDefinitions.RootSquashModeOption);
+        var noSquashNidLists = commandResult.GetValueOrDefault(AzureManagedLustreOptionDefinitions.NoSquashNidListsOption);
+        var squashUid = commandResult.GetValueOrDefault(AzureManagedLustreOptionDefinitions.SquashUidOption);
+        var squashGid = commandResult.GetValueOrDefault(AzureManagedLustreOptionDefinitions.SquashGidOption);
+
+
+        // If root squash mode is provided and not 'none', require UID, GID and no squash NID list
+        if (!string.IsNullOrWhiteSpace(rootSquashMode) && !rootSquashMode.Equals("None", StringComparison.OrdinalIgnoreCase))
+        {
+            if (!(squashUid.HasValue && squashGid.HasValue && !string.IsNullOrWhiteSpace(noSquashNidLists)))
+            {
+                if (commandResponse is not null)
+                {
+                    commandResponse.Status = 400;
+                    commandResponse.Message = "When --root-squash-mode is not 'None', --squash-uid, --squash-gid and --no-squash-nid-list must be provided.";
+                }
+                return new ValidationResult { IsValid = false, ErrorMessage = commandResponse?.Message };
+            }
+        }
+
+        return new ValidationResult { IsValid = true };
+
+    }
+
+    public ValidationResult ValidateMaintanenceOptionsCreate(CommandResult commandResult, CommandResponse? commandResponse)
+    {
+        return ValidateMaintenanceOptions(commandResult, commandResponse, false);
+    }
+
+    public ValidationResult ValidateMaintanenceOptionsUpdate(CommandResult commandResult, CommandResponse? commandResponse)
+    {
+        return ValidateMaintenanceOptions(commandResult, commandResponse, true);
+    }
+    public ValidationResult ValidateMaintenanceOptions(CommandResult commandResult, CommandResponse? commandResponse = null, bool update = false)
```

Comment:
Since these all are public, I'd just have the callers pass `update` as appropriate